### PR TITLE
Fixes #33782 - do not use fixtures for Settings

### PR DESCRIPTION
**Pull request by ezr-ondrej**
As we no longer require settings in DB, we may drop the fixtures for settings.

This should wait for ~#9062~ and ~#9068~

### CODE REVIEW

File: test/fixtures/settings.yml [hunk 1]
```diff
@@ -1,402 +1,4 @@
...
-  name: default_host_init_config_template
-  category: Setting
-  default: 'Linux host initial configuration'
-  description: "Default host initial configuration template"
-attribute96:
-  name: server_ca_file
-  category: Setting
-  default: /var/lib/puppet/ssl/certs/ca.pem
-  description: SSL CA file that will be used in templates (to verify the connection to Foreman)
 attribute97:
```
Comment (ekohl) [inline test/fixtures/settings.yml:8]: Are these here because the override the default?

Comment (ezr-ondrej) [reply test/fixtures/settings.yml:8]: yeah, we need a different value then a default. I'm not happy about it, I do not agree with a bin path being in settings, but I do not have better solution for it.

**Comment by lzap** [comment]
Where are these values now? I mean, I am used to these values are always available in tests.

**Comment by ezr-ondrej** [comment]
> Where are these values now? I mean, I am used to these values are always available in tests.

Those were just defaults and those now come from the settings definitions. We do not need them in the database anymore, unless we need a different value in the test setup, which we should not need unless specifically stated in the test itself IMHO

**Comment by lzap** [comment]
Ok once this is green let me know, I will test and fix my plugins. I am sure I use fixtures, if i do, let's warn other authors. Maybe I am just suspicious incorrectly.

**Comment by ekohl** [comment]
In Django I always liked the [override_settings](https://docs.djangoproject.com/en/4.0/topics/testing/tools/#django.test.override_settings) helper. Having a formal API in your test suite to say you want to test with certain settings was incredibly useful, even though nowadays I hardly interact with Django. Perhaps we can implement a similar API?

**Comment by ezr-ondrej** [comment]
> Perhaps we can implement a similar API?

I'll not get around to that, it would be nice but I think that `Setting['setting_name'] = value` at the top of the test is not that bad and I'll do so for the failing tests.

**Comment by ezr-ondrej** [comment]
This feels ready :thinking:

**Review by ekohl — APPROVED** [review]

**Pull request merged**